packages/edit-site/src/components/block-editor/resizable-editor.js

```diff
@@ -182,7 +182,14 @@ function ResizableEditor( { enableResizing, settings, ...props } ) {
 				name="editor-canvas"
 				className="edit-site-visual-editor__editor-canvas"
 				{ ...props }
-			/>
+			>
+				<span
+					dangerouslySetInnerHTML={ {
+						__html: window._wpSvgFilters,
+					} }
+				/>
```

Comment:
@oandregal (re: https://github.com/WordPress/gutenberg/pull/37727#discussion_r789516704)

> Can we enqueue the default and theme SVGs from the server instead? The only change in the client would be to create the selectors using the duotone selector.

The filters still have to make their way into the iframe in the body before the content, but I was able to make it happen by throwing them into a variable and injecting them here. It feels a little awkward, but it avoids the JS `addFilter`.

I'm still not very excited about all of the changes needed for adding the styles, especially the `valueFunc` function that is going to be easy to forget if it ever changes in the PHP. Maybe we could do something similar for the styles by rendering them all on the server and injecting them like this.

Reply:
I'm not sure there's a better way to load the SVGs, I'm going to ping @ellatrix for thoughts.

Ella, for context: the duotone feature needs to enqueue a bunch of `SVG` HTML elements in the scope of the content, as they're used as filter styles. This PR explores make them available via the `_wpSvgFilters` global. We use `wp_body_open` for the frontend. Is there a better alternative?

Reply:
Didn't we solve this previously with a portal? 

https://github.com/WordPress/gutenberg/blob/d06ee3a4c5d18e336e2a7f3e78dcf6a7f9a6e397/packages/block-editor/src/hooks/duotone.js#L294-L309

Reply:
@ellatrix That loads the per-block overrides, not the global styles filters. For global styles, the filters get rendered in the `wp_body_open` hook here https://github.com/WordPress/gutenberg/blob/42cd5bbded43f84224c6d18e1c8f5aacae8e1268/lib/compat/wordpress-5.9/render-svg-filters.php#L16-L26

Reply:
And we can't do the same thing with `useContext( BlockList.__unstableElementContext )`?

Reply:
No, because that is for the per-block overrides, not the global styles.

Take a look at the branch [one commit before the last](https://github.com/WordPress/gutenberg/compare/trunk...59b4151e55f5ee2972b87e06cffd4cf5546ba649) where I previously added a place for rendering the SVGs in JS to get an idea as to what this current approach is an alternative to.